custom_components/frigate/media_source.py

```diff
@@ -437,13 +437,35 @@ def from_str(
         if len(parts) < 2 or parts[1] != cls.get_identifier_type():
             return None
 
+        print(f"The parts are {parts}")
+        tz = "UTC" if (len(parts) == 6 and str(parts[2]).count("-") == 2) else "LOCAL"
```

Comment:
Two recommendations:
 - Add a comment explaining that pre $version_number Frigate used %Y-%m and post $version_number uses %Y-%m-%d *and* the latter implies it will also use UTC.
 - Do a complete validation of the string rather than simply counting the hashes, e.g.

tz = "UTC"
try:
  datetime.datetime.strptime(parts[2], '%Y-%m-%d')
except ValueError:
  tz = "LOCAL"

Perhaps something like this is more verbose, but it's also better validation and more readable as to what is really going on?

Reply:
yeah makes sense, makes it more understandable what is actually going on